File: src/main/java/ru/timeconqueror/timecore/common/capability/property/serializer/BooleanPropertySerializer.java
package ru.timeconqueror.timecore.common.capability.property.serializer;

import net.minecraft.nbt.CompoundTag;

public class BooleanPropertySerializer implements IPropertySerializer<Boolean> {
    public static final IPropertySerializer<Boolean> INSTANCE = new BooleanPropertySerializer();
    public static final IPropertySerializer<Boolean> NULLABLE = IPropertySerializer.nullable(INSTANCE);

    @Override
    public void serialize(String name, Boolean value, CompoundTag nbt) {
        nbt.putBoolean(name, value);
    }

    @Override
    public Boolean deserialize(String name, CompoundTag nbt) {
        return nbt.getBoolean(name);
    }


}

File: src/main/java/ru/timeconqueror/timecore/common/capability/property/serializer/BooleanPropertySerializer.kt
package ru.timeconqueror.timecore.common.capability.property.serializer

import net.minecraft.nbt.CompoundTag


object BooleanPropertySerializer : IPropertySerializer<Boolean> {

    override fun serialize(name: String, value: Boolean, nbt: CompoundTag) = nbt.putBoolean(name, value)
    override fun deserialize(name: String, nbt: CompoundTag) = nbt.getBoolean(name)

    object Nullable : NullPropertySerializer<Boolean>(this)
}